src/_path.h

```diff
@@ -279,8 +279,8 @@ inline bool point_on_path(
     point.push_back(y);
     points.push_back(point);
 
-    std::vector<bool> result(1);
-    result[0] = false;
+    int result[1];
```

Comment:
This is all very subtle (which is why it got lost).  If this were `std::vector<int>`, then there is no guarantee that it gets passed as a `const &std::vector<int>` to `point_in_path_impl` (where the parameter is templatized), so `point_in_path_impl` ends up storing the results in a temporary array that is thrown away rather than where you want to write it to.  (It probably could be made to, but it requires some template specialization techniques that proved not to be portable).  This need for using a C array also means we can't query its size in the loops you mention above, but these are all internally created arrays (not passed in from the Python side), so the potential for a length mismatch is small.